src/plugins/workspace/server/plugin.ts

```diff
@@ -55,12 +60,98 @@ export class WorkspacePlugin implements Plugin<{}, {}> {
     });
   }
 
+  private isRequestByDashboardAdmin(
+    request: OpenSearchDashboardsRequest,
+    groups: string[],
+    users: string[],
+    configGroups: string[],
+    configUsers: string[]
+  ) {
+    if (configGroups.length === 0 && configUsers.length === 0) {
+      updateWorkspaceState(request, {
+        isDashboardAdmin: false,
+      });
+      return;
+    }
+    const groupMatchAny = groups.some((group) => configGroups.includes(group)) || false;
+    const userMatchAny = users.some((user) => configUsers.includes(user)) || false;
+    updateWorkspaceState(request, {
+      isDashboardAdmin: groupMatchAny || userMatchAny,
+    });
```

Comment:
From the function name, I do not think it appropriate to `update` anything inside this function.

Reply:
Sure, I will change the function name to `updateDashboardAdminStateForRequest`